projects/RabbitMQ.Client/client/impl/AsyncConsumerWorkService.cs

```diff
@@ -83,9 +101,55 @@ async Task Loop()
                 }
             }
 
+            async Task LoopWithConcurrency(CancellationToken cancellationToken)
+            {
+                try
+                {
+                    while (await _channel.Reader.WaitToReadAsync(cancellationToken).ConfigureAwait(false))
+                    {
+                        while (_channel.Reader.TryRead(out Work work))
+                        {
+                            // Do a quick synchronous check before we resort to async/await with the state-machine overhead.
+                            if(!_limiter.Wait(0))
+                            {
+                                await _limiter.WaitAsync(cancellationToken).ConfigureAwait(false);
+                            }
+
+                            _ = HandleConcurrent(work, _model, _limiter);
```

Comment:
<3 the clarity of intention by using the discard :)

Reply:
Some teams have extensions methods like

```
public static void IntentionallyNotAwaited|Ignored(this Task task) {
 //
}
```

but given we have the language constructs I think discard is fine. Glad you like it ;)